### PR TITLE
feat: added dask `Expr.__sub__`

### PR DESCRIPTION
**Pull request by anopsy**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637 
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

Hey @MarcoGorelli what about the tests for dunder methods - I noticed there's no specific test for `Expr.__add__` or can we treat the with_column test as such? 
edit: Okay I figured it out. test_with_columns indeed covers for add/sub/mul, cool!

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
perfect, thanks @anopsy !

feel free to merge after resolving the merge conflicts

**Comment by anopsy** [comment]
tox check is again unhappy, looks like a different type of import problems, still seems unrelated so I'll merge ;)

File: narwhals/dependencies.py [hunk 1]
```diff
@@ -72,7 +73,17 @@ def get_dask() -> Any:
...
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message="\nDask dataframe query planning.*",
+                category=FutureWarning,
+            )
+            import dask.dataframe as dd
+
+        return dd
+    return None  # pragma: no cover
```
Comment (MarcoGorelli) [inline narwhals/dependencies.py:86]: could you revert this change please? that should fix the tox issue

Comment (anopsy) [reply narwhals/dependencies.py:86]: Yes! I need to dive into it, will finish tomorrow morning 🫡

Comment (MarcoGorelli) [reply narwhals/dependencies.py:86]: cool, thanks! it's only the highlighted change that needs reverting, the rest is fine 😇 

I suspect something went not quite right when resolving merge conflicts

no worries, this is how we learn 😄 🤗

Comment (anopsy) [reply narwhals/dependencies.py:86]: Yes sorry for that, I deleted somehow one of the parentheses while removing the markers from merge conflict, and when I was fixing it, I saw that I have unstaged changes to dependencies.py file,  I did add/commit only my changes but this file somehow ended up in the pull request, but that's prbably because of merging main into my branch (if I understand it correctly). I'm looking now how to proceed with reverting. I already clicked on the "revert" button. I see that it added a commit. Is this the same as doing "git revert -m 1 a9c34dd"

Comment (MarcoGorelli) [reply narwhals/dependencies.py:86]: 😄 no worries at all (and no need to apologise)

I wouldn't do this automatically - git revert is for reverting the whole PR, rather than just a part of it

in this particular case, I ended up changing it as part of another PR, so nothing left to do 😇 

In the future, before merging, I'd suggest clicking "files changed" on the PR, refreshing the page, and then checking that there's nothing there which you didn't explicitly intend to add (if there is, it means something might not have gone right, e.g. when merging)
